Author: Jakub Wilk <jwilk@debian.org>
Last-Update: Tue, 20 May 2014 13:35:26 +0200
Bug-Debian: http://bugs.debian.org/733407
Description: this is quick and dirty patch to use %parse-param instead
 of deprecated YYPARSE_PARAM (and eventually removed in Bison 3.0).
 %parse-param was added in bison 1.875, over a decade ago, so no
 Build-Depends adjustments are needed. :-)

--- a/src/Mixfix/top.yy
+++ b/src/Mixfix/top.yy
@@ -24,6 +24,8 @@
 //	Parser for Maude surface syntax.
 //
 
+%parse-param {void* YYPARSE_PARAM}
+
 %{
 #include <string>
 #include <stack>
@@ -91,7 +93,7 @@ SyntaxContainer* oldSyntaxContainer = 0;
 Int64 number;
 Int64 number2;
 
-static void yyerror(char *s);
+static void yyerror(void *, char *s);
 
 void cleanUpModuleExpression();
 void cleanUpParser();
--- a/src/Mixfix/bottom.yy
+++ b/src/Mixfix/bottom.yy
@@ -23,7 +23,7 @@
 %%
 
 static void
-yyerror(char *s)
+yyerror(void *, char *s)
 {
   if (!(UserLevelRewritingContext::interrupted()))
     IssueWarning(LineNumber(lineNumber) << ": " << s);
